src/dialer/index.js

```diff
@@ -74,20 +74,38 @@ class Dialer {
     if (!dialTarget.addrs.length) {
       throw errCode(new Error('The dial request has no addresses'), codes.ERR_NO_VALID_ADDRESSES)
     }
-    const pendingDial = this._pendingDials.get(dialTarget.id) || this._createPendingDial(dialTarget, options)
+
+    // Used for subsequent dials pending
+    let subsequentDialAborted = false
+    const onAbort = () => {
+      subsequentDialAborted = true
+      pendingDial.controller.abort()
+    }
+
+    let pendingDial = this._pendingDials.get(dialTarget.id)
+    if (!pendingDial) {
+      pendingDial = this._createPendingDial(dialTarget, options)
+    } else {
+      // track subsequent dial abort
+      options.signal && options.signal.addEventListener('abort', onAbort)
+    }
```

Comment:
I'm going to look more at the underlying issue. 

I don't think this is the right way to solve this problem and it's likely to have some unwanted side effects. If we assume an internal system like the DHT is actively trying to dial a peer, if the application has a shorter timeout tolerance, and dials and aborts before the internal DHT dial is done it will terminate the underlying dial which is not good. We need to be able to cancel the existing context without affecting any dials that are already in progress.

Reply:
My main concern in this issue is how to guarantee that we do not reach unexpected states.
As an example, if the application layer performs a dial and it is aborted, but then a parallel dial connection was achieved, does this seem correct? I understand the possible side effects for other subsystems, but should a connection exist after an application layer abort? I don't know, but if we do not abort other dials, the abort signal option should be clear on what it does.

Reply:
Abort signals should only apply to their current context. If aborting can affect other contexts it would make unexpected states much worse, as you can get a lot of really nasty collisions between the subsystems and application layer. While it might be odd to have a connection established when you aborted a dial, the circumstances of how that parallel dial can vary quite a bit, and aborting the current context is not the right way to manage that kind of behavior if that's what the user needs.